CoreScriptsRoot/Modules/BackpackScript3D.lua

```diff
@@ -188,12 +192,22 @@ local function AddTool(tool)
 
 	slot.hovered = false
 	slot.tool = tool
-	slot.icon = Instance.new("ImageButton", toolsFrame)
+	if tool.TextureId ~= "" then
```

Comment:
Not a bad idea, but I'll wrap them both in a TextButton so nothing else has to change.